packages/react-cosmos/src/shared/devServer/httpServer.ts

```diff
@@ -1,17 +1,62 @@
-import http from 'http';
+import http, { Server } from 'http';
+import https from 'https';
+import fs from 'fs';
+const pem = require('pem');
 import { CosmosConfig } from '../../config';
 
 type RequestListener = (
   request: http.IncomingMessage,
   response: http.ServerResponse
 ) => void;
 
-export function createHttpServer(
+export async function createHttpServer(
   cosmosConfig: CosmosConfig,
   requestListener: RequestListener
 ) {
-  const { port, hostname } = cosmosConfig;
-  const server = http.createServer(requestListener);
+  const { port, hostname, https: httpsEnabled, httpsOptions } = cosmosConfig;
+
+  let server: Server;
+  if (httpsEnabled) {
+    let credentials: {
+      key: string;
+      cert: string;
+    };
+
+    if (httpsOptions?.key && httpsOptions?.cert) {
+      credentials = {
+        key: fs.readFileSync(httpsOptions.key, 'utf8'),
+        cert: fs.readFileSync(httpsOptions.cert, 'utf8'),
```

Comment:
These are file paths, right? The user should be able to use relative paths like `mycert.crt` or `./mycert.crt`, like all the other paths are handled in the Cosmos config.

Here are two examples of how paths are handled currently, basically resolving them from `rootDir`:
https://github.com/react-cosmos/react-cosmos/blob/1a54f3cf37afd0440d05a7a06897994bdbf56712/packages/react-cosmos/src/config/createCosmosConfig.ts#L27-L35

I'm even thinking of naming the options `keyPath` and `certPath` for both consistency and clarity of their value.

Reply:
made changes